make es6ify idempotent

hkollmann opened the pull request:
The problem is the call of babelCore.transform. This will change output during repeated calls.
Implement an modfied idea from @derrel. 

while (true) {
        result = babelCore.transform(src, config);
        if (result.code === src) {
          break;
        }
        src = result.code;
      }

hkollmann committed:
make es6ify idempotent

johnspackman reviewed (approved):
Brilliant :)

Approving, just wondered if it works on the pre-es6ified source?

derrell commented on source/class/qx/tool/compiler/Es6ify.js:
```diff
@@ -168,7 +168,14 @@ qx.Class.define("qx.tool.compiler.Es6ify", {
+          break;
+        }
+        src = result.code;
+      }
```
This looks good... but is there any chance that it could get into a state where it toggles back and forth between two results, or otherwise doesn't stabilize? Maybe add a maximum number of iterations (10? 50?) as a "just in case"? I'll approve this, but I do have a bit of concern.

derrell replied:
Maybe even add a printed warning if it ever does hit the maximum count, so that we can look into what's causing it?

oetiker replied:
yep, having an escape hatch sounds like a good thing ...

oetiker commented on source/class/qx/tool/compiler/Es6ify.js:
```diff
@@ -168,7 +168,14 @@ qx.Class.define("qx.tool.compiler.Es6ify", {
 
-      let result = babelCore.transform(src, config);
+      let result;
+      while (true) {
```
```suggestion
      let cycleCount = 0;
      while (true) {
```

derrell replied:
lol. I'll assume you pressed the green button before you were finished typing in your suggestion...

hkollmann commented:
@johnspackman : Yes it did. Tested with framework code.

hkollmann commented:
Thought about an exit strategy after launching the PR...

hkollmann committed:
add exit strategy
use standard qx.tool.compiler.Console.info instead of console log
remove babel warning

hkollmann commented:
After merging this we have rerun the es6ify process...
